### PR TITLE
Add the test input to zero tests on submission details page

### PR DESCRIPTION
**Pull request by msmner**
Closes 
https://github.com/SoftUni-Internal/exam-systems-issues/issues/760
https://github.com/SoftUni-Internal/exam-systems-issues/issues/785
https://github.com/SoftUni-Internal/exam-systems-issues/issues/784

Linked pull requests from another repos (if any):
https://github.com/SoftUni-Internal/judge-submissions-distributor/pull/18

**Summary of the changes made**:
1. Add show input collapsible to submission details page for zero tests and for admins/lecturers or if show details is true
2. Add execution comment to submission details page

### CODE REVIEW

**Review by shsimeonova — DISMISSED** [review]

**Review by shsimeonova — APPROVED** [review]

File: Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs [hunk 1]
```diff
@@ -99,6 +104,26 @@ public async Task<SubmissionDetailsServiceModel> GetDetailsById(int submissionId
             throw new BusinessServiceException(validationResult.Message);
         }
 
+        var contest = await this.contestsDataService.GetByProblemId<ContestServiceModel>(submissionDetailsServiceModel!.Problem.Id).Map<Contest>();
+        var userIsAdminOrLecturerInContest = currentUser.IsAdmin || IsUserLecturerInContest(contest, currentUser.Id!);
+        var showTestInputForAllTests = submissionDetailsServiceModel.Problem.ShowDetailedFeedback;
+        if (!userIsAdminOrLecturerInContest && !showTestInputForAllTests)
+        {
+            submissionDetailsServiceModel.TestRuns = submissionDetailsServiceModel.TestRuns.Select(x =>
```
Comment (PetarNeshkov) [inline Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs:112]: In order the code to be more readable, you can change this to `.Select(tr =>
            {
                if (!tr.IsTrialTest)
                {
                    tr.ShowInput = false;
                    tr.Input = string.Empty;
                    tr.ExecutionComment = string.Empty;
                    tr.ExpectedOutputFragment = string.Empty;
                    tr.UserOutputFragment = string.Empty;
                }
                return tr;
            });
        }`

**Review by PetarNeshkov — DISMISSED** [review]

File: Services/UI/OJS.Services.Ui.Models/Contests/ContestServiceModel.cs [hunk 2]
```diff
@@ -209,5 +209,6 @@ public bool ResultsArePubliclyVisible
             .ForMember(d => d.UserIsAdminOrLecturerInContest, opt => opt.Ignore())
             .ForMember(d => d.UserCanCompete, opt => opt.Ignore())
             .ForMember(d => d.UserIsParticipant, opt => opt.Ignore())
-            .ForMember(d => d.IsActive, opt => opt.Ignore());
+            .ForMember(d => d.IsActive, opt => opt.Ignore())
+            .ReverseMap();
```
Comment (PetarNeshkov) [inline Services/UI/OJS.Services.Ui.Models/Contests/ContestServiceModel.cs:213]: What is the benefit of using .ReverseMap() in this particular case?

File: Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs (hunk 1 above)
Comment (msmner) [reply Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs:112]: Done.

File: Services/UI/OJS.Services.Ui.Models/Contests/ContestServiceModel.cs (hunk 2 above)
Comment (msmner) [reply Services/UI/OJS.Services.Ui.Models/Contests/ContestServiceModel.cs:213]: I need the contest for the method IsUserLecturerInContest. The service model does not have the properties that are needed to check if the user is lecturer in the contest. The easiest way to get the contest was just to map the service model to the contest on line 107: 
var contest = await this.contestsDataService.GetByProblemId<ContestServiceModel>(submissionDetailsServiceModel!.Problem.Id).Map<Contest>();

**Review by PetarNeshkov — APPROVED** [review]
